src/emailagent.cpp

```diff
@@ -179,9 +179,17 @@ quint64 EmailAgent::downloadMessagePart(const QMailMessagePart::Location &locati
     return enqueue(new RetrieveMessagePart(m_retrievalAction.data(), location, false));
 }
 
-void EmailAgent::exportUpdates(const QMailAccountId accountId)
+void EmailAgent::exportUpdates(const QMailAccountIdList accountIdList)
 {
-    enqueue(new ExportUpdates(m_retrievalAction.data(),accountId));
+    if (!m_enqueing && accountIdList.size()) {
```

Comment:
All this fiddling about with m_enqueing needs to be fixed. It seems to just be preventing enqueued actions from being initiated - I think it would be clearer to replace m_enqueing with a boolean argument added to the enqueue(action, startIfNotBusy) method.